src/run/port.rs

```diff
@@ -1,19 +1,21 @@
+use crate::ops::Num;
+
 use super::*;
 
 /// A port in the interaction net.
 ///
 /// The type of a port is determined by its *tag*, which is stored in the bottom
 /// three bits.
 ///
-/// All tags other than [`Num`] divide the bits of the port as follows:
+/// All tags other than [`Int`] divide the bits of the port as follows:
 /// - the top 16 bits are the *label*, accessible with [`Port::lab`]
 /// - the middle 45 bits are the non-alignment bits of the *address*, an
 ///   8-byte-aligned pointer accessible with [`Port::addr`]
 /// - the bottom 3 bits are the tag, as always
 ///
 /// The semantics of these fields depend upon the tag; see the documentation for
 /// each [`Tag`] variant.
-#[derive(Clone, Eq, PartialEq, PartialOrd, Hash, Default)]
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Hash, Default)]
```

Comment:
this is intentionally not `Copy`; a safe API will necessitate cloning of ports being `unsafe`, so having those `.clone()` calls be explicit is desired